test/jasmine/tests/config_test.js

```diff
@@ -560,17 +579,60 @@ describe('config argument', function() {
         }
 
         it('should resize when the viewport width/height changes', function(done) {
-            var newWidth = 400, newHeight = 700;
-            Plotly.newPlot(gd, data, {}, {responsive: true})
+            var promise = Plotly.plot(gd, data, {}, {responsive: true});
+
+            testResponsive(promise)
+            .then(done);
+        });
+
+        it('should still be responsive if the plot is redrawn', function(done) {
+            var promise = Plotly.plot(gd, data, {}, {responsive: true})
+            .then(function() {
+                Plotly.restyle(gd, 'y[0]', data[0].y[0] * 2);
```

Comment:
You need to `return` these `Plotly.*` calls inside `.then` - as with any function that itself returns a promise - so the new promise gets added to the chain.